ui/src/app/components/add-index-form/add-index-form.component.spec.ts

```diff
@@ -1,36 +1,192 @@
+      providers: [
+        FormBuilder,
+        { provide: DataService, useValue: dataServiceSpy },
+        { provide: SidenavService, useValue: sidenavServiceSpy },
+        { provide: ConversionService, useValue: conversionServiceSpy },
+      ],
     })
-    .compileComponents();
+      .compileComponents();
+
+    fb = TestBed.inject(FormBuilder);
+
+    conversionServiceSpy.getColIdFromSpannerColName.and.returnValue("TestId")
+    conversionServiceSpy.getTableIdFromSpName.and.returnValue("t1")
+    dataServiceSpy.conv = of(mockIConv);
+    dataServiceSpy.dropRule.and.callThrough();
+    sidenavServiceSpy.sidenavAddIndexTable = of("t1");
+    sidenavServiceSpy.displayRuleFlag = of(true)
+    sidenavServiceSpy.ruleData = of()
   });
 
   beforeEach(() => {
     fixture = TestBed.createComponent(AddIndexFormComponent);
     component = fixture.componentInstance;
+    component.addIndexForm = fb.group({
+      tableName: [''],
+      indexName: ['', []],
+      ColsArray: fb.array([]),
+    });
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should render ngOnInit and all dependent services correctly', () => {
+    const addIndexRule: IRule = {
+      Id: "r1",
+      Name: "x1",
+      ObjectType: "Table",
+      Type: "add_index",
+      Enabled: true,
+      Data: {
+        Id: "ind1",
+        Name: "ind1",
+        TableId: "t1",
+        Keys: [
+          {
+            ColId: "c1",
+            Order: 1,
+            Desc: false,
+          }
+        ]
+      }
+    }
+    sidenavServiceSpy.ruleData = of(addIndexRule)
+    component.ngOnInit()
+    expect(component.ColsArray.length).toBe(1);
+    expect(component.ColsArray.at(0).value.columnName).toEqual('column1');
+    expect(component.addIndexForm.controls['tableName'].value).toEqual("table1");
+    expect(component.addIndexForm.controls['indexName'].value).toEqual("ind1")
+    expect(component.addIndexForm.status).toEqual("DISABLED");
+  });
+
+  it('should add a new column form', () => {
+    component.addNewColumnForm();
+    expect(component.ColsArray.length).toBe(1);
+  });
+
+  it('should remove a column form', () => {
+    component.addNewColumnForm();
+    component.addNewColumnForm();
+    component.removeColumnForm(1);
+    expect(component.ColsArray.length).toBe(1);
+  });
+
+  it('should reset rule type', () => {
+    spyOn(component.resetRuleType, 'emit');
+    component.resetRuleType.emit('');
+    expect(component.resetRuleType.emit).toHaveBeenCalledWith('');
+  });
+
+  it('should set column arrays for view rules', () => {
+    const tableId = 't1';
+    const ruleData = {
+      Keys: [
+        {
+          ColId: 'c1',
+          Desc: false,
+          Order: 1,
+        },
+      ],
+    };
+    component.setColArraysForViewRules(tableId, ruleData.Keys);
+    expect(component.ColsArray.length).toBe(1);
+    expect(component.ColsArray.at(0).value.columnName).toEqual('column1');
```

Comment:
Assert using the input test data, `column1` is not mentioned in `ruleData` or `tableId`.